Fix non terminal error

JPWerling opened the pull request:
## Contents


### The What
A fix to the package not being able to run in non-terminal instances, such as cronjobs.

### The Why
Enables running the package non-terminal

### The How
Adds a check for connected terminal before trying to access attributes about it.

### This [update](https://semver.org/) is:
- [x] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Test Procedure

### Installation and initiation
* `git clone git@github.com:genomic-medicine-sweden/NGPIris.git`
* `cd NGPIris && git checkout <BRANCH>`
* `bash setup.sh`
* `source activate hpcenv`

### Tests
* `pytest tests/`
* Potential additional tests

### Expected outcome:
* Pytest resolves without crashes
* Potential additional results

## Confirmations:
- [ ] Code reviewed by @octocat
- [ ] Code tested by @octocat

JPWerling committed:
Add error raise for dynamically reading large file

Raise error rather than return empty string

JPWerling committed:
Fix trying to access terminal when none connected

Upload/download progress printing attempted to get the size of the
connected terminal, resulting in errors for when no terminal was
connected, in such cases as running via cronjobs.

Add check for connected terminal and return empty string if none found.

JPWerling committed:
Add context managers to ProgressPercentage invokes

Move printing of average speeds to __exit__ dunder.

Remove callback kwarg from upload and download, since there was little
incentive to use it. Defaults to always using the upload and download
callback. Could be implemented back as a kwarg properly at a later time.

Fix/remove various syntax issues and unused statements.

JPWerling committed:
Update version

sylvinite commented on NGPIris/hcp/hcp.py:
```diff
@@ -216,30 +219,14 @@ def search_objects(self, string):
 
     @bucketcheck
-    def upload_file(self, local_path, remote_key, metadata={},callback="default"):
+    def upload_file(self, local_path, remote_key, metadata={}):
```
Refering to our previous conversation.
Are you certain this works? Afaik pytests crash if `callback = ProgressPercentage(local_path)` is used.

JPWerling replied:
Removed left-over references to the callback kwarg and all tests pass

sylvinite reviewed (changes requested):
Most is agreeable. See single comment.

JPWerling committed:
Remove reference to download/upload callback kwarg

JPWerling committed:
Add enable/disable of upload/download callback

JPWerling committed:
Add callback silencing functionality to README

JPWerling committed:
Fix generate_tagmap creating broken json files

JPWerling committed:
Add random string to metadata.json during tests

The metadata.json generated in /tmp/ could cause permission issues when
different users tested the code on the same system.

JPWerling committed:
Disable callback when silent mode selected via CLI
